### PR TITLE
Replaced KnpPaginator by PagerFanta

### LINKED ISSUE #305: Replace KnpPaginatorBundle by PagerfantaBundle
KnpPaginatorBundle has some bad parts:

- It doesn't support the translation https://github.com/KnpLabs/KnpPaginatorBundle/issues/334
- It has a bug with the pagination https://github.com/KnpLabs/knp-components/issues/78
- IMO: it has a very ugly design

I tried to fix the first issue here https://github.com/symfony/symfony-demo/pull/304, but I think it's better to get the rid of KnpPaginatorBundle at all.

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #305.

### CODE REVIEW

File: src/AppBundle/Controller/BlogController.php [hunk 1]
```diff
@@ -34,23 +34,15 @@ class BlogController extends Controller
 {
     /**
      * @Route("/", defaults={"page": 1}, name="blog_index")
-     * @Route("/page/{page}", requirements={"page": "[1-9]\d*"}, name="blog_index_paginated")
+     * @Route("/page/{page}", name="blog_index_paginated")
```
Comment (voronkovich) [inline src/AppBundle/Controller/BlogController.php]: @javiereguiluz, why did you delete the requirements section?

Comment (javiereguiluz) [inline src/AppBundle/Controller/BlogController.php]: No valid reason ... so I've restored it. Thanks!

**Comment by voronkovich** [comment]
:+1:

File: web/bundles/whiteoctoberpagerfanta [hunk 2]
```diff
@@ -0,0 +1 @@
+../../vendor/white-october/pagerfanta-bundle/Resources/public/
```
Comment (xabbuh) [inline web/bundles/whiteoctoberpagerfanta]: This looks weird.

**Comment by xabbuh** [comment]
Are the changes in `web/config.php` in any way related to this change?

**Comment by javiereguiluz** [comment]
@xabbuh to be honest ... I don't know. I've reverted everything and we can update the `web/config.php` in a separate PR.

**Comment by xabbuh** [comment]
:+1:

File: src/AppBundle/Repository/PostRepository.php [hunk 3]
```diff
@@ -36,8 +43,17 @@ public function queryLatest()
...
-    public function findLatest()
+    /**
+     * @param int $page
+     *
+     * @return Pagerfanta
+     */
+    public function findLatest($page = 1)
     {
-        $this->queryLatest()->getResult();
+        $paginator = new Pagerfanta(new DoctrineORMAdapter($this->queryLatest(), true, false));
```
Comment (stof) [inline src/AppBundle/Repository/PostRepository.php]: technically, there is no fetch joined collection in this query, so you can optimize the pagination by changing the second argument

Comment (javiereguiluz) [inline src/AppBundle/Repository/PostRepository.php]: You're right. Done. Thanks.
